core/lib/types/src/block.rs

```diff
@@ -91,6 +91,7 @@ pub struct L2BlockHeader {
     /// amount of gas can be spent on pubdata.
     pub gas_limit: u64,
     pub logs_bloom: Bloom,
+    pub pubdata_params: PubdataParams,
```

Comment:
Dumb question: Can `pubdata_params` be changed in a middle of batch? (Looks like not, judging by the current implementation.) Or are they present in blocks rather than L1 batches because of the issue with storing these params for a pending batch? In any case, it looks misleading to place these params in `SystemEnv` (vs `L1BatchEnv`) because they can change per L1 batch without any sort of coordination (vs protocol version or base system contracts).

Reply:
Yes, they're present in blocks because of the pending batch issue. I hope that `pending_batch` will be in DB really soon and then we can remove pubdata_params from miniblocks.
Also, extending Stan's answer, it requires coordination but it's not a protocol version parameter. It's not decided how that coordination will look like nor implemented yet, effectively it means that operator is not allowed to change `pubdata_params` for now